packages/lodestar/src/chain/chain.ts

```diff
@@ -110,6 +110,19 @@ export class BeaconChain extends (EventEmitter as { new(): ChainEventEmitter })
     return this.db.block.get(summary.blockRoot);
   }
 
+  public async getUnfinalizedBlocksAtSlots(slots: Slot[]): Promise<SignedBeaconBlock[]> {
+    if (!slots) {
+      return [];
+    }
+    const blockRoots = slots.map((slot) => {
+      const summary = this.forkChoice.getBlockSummaryAtSlot(slot);
+      return summary? summary.blockRoot : null;
+    }).filter((blockRoot) => !!blockRoot);
+    // these blocks are on the same chain to head
```

Comment:
Since we get block roots from the same head, we're sure all blocks are on the same chain to head